### ISSUE TITLE
SHgrid.expand().expand()

### ISSUE THREAD

**Issue by Totadarck** [SHTOOLS/SHTOOLS#325]
Hello everyone,
I'm starting with spherical harmonics. So I don't know if it's a problem with my inputs or my code. I was just trying to test transforming a grid into a harmonic coefficient and then converting the coefficients back into a grid. The problem I have is that the grid that comes out is different from the input grid. Can someone help me? Excuse me also if this is not the place to ask this kind of question. 

import matplotlib.pyplot as plt
import pyshtools as pysh
import numpy as np

test=np.array([[1,2,4,2],[3,1,6,8]])
gridt=pysh.SHGrid.from_array(test,'GLQ')
coefft=pysh.SHCoeffs.from_array(gridt.expand().coeffs)
grid2t=coefft.expand('GLQ').to_array() 
plt.pcolor(grid2t-test)
plt.colorbar()
plt.show()

Have a nice day 

HENRY Adrien

windows 10 x64

**Comment by MarkWieczorek** [comment]
The nlat=2, long=4 GLQ grid that you are using can resolve functions expressed in spherical harmonic degrees up to 1. I suspect that the arbitrary function you created has components with degrees greater than 1, and hence the spherical harmonic transform has aliasing problems.

For tests, you might want to create a set of spherical harmonic coefficients first, and then expand in onto a grid, as opposed to the other way around.

**Comment by MarkWieczorek** [comment]
Also, for GLQ grids of this size, the first and last longitude correspond to 0 and 360 degrees, and thus the values should be the same.

**Comment by Totadarck** [comment]
Hello and thank you for this answer.

As I said, I am a novice in the field of spherical harmonics. Indeed, I had not yet integrated that the conversion in spherical harmonic coefficient was an approximation of the measurements by the harmonic function. I am working on the resolution of the sea level equation for my thesis. I corrected the input grid to make the first and last column equal.

I have been experimenting with decreasing the input grid resolution (interpolation) while keeping the maximum degree high. In doing so, I noticed that the minimum and maximum error was lower for a grid resolution equal to the maximum degree multiplied by twice the maximum degree. I also noticed that the standard deviation of the error was low for this grid resolution (it is not the lowest). Finally, I noticed that the standard deviation of the error was lower and lower the more one increases the maximum degree (which is logical).


However, I was wondering if there were other methods to decrease the error?
